services/cloudapi/cloudapinic/nic.go

```diff
@@ -0,0 +1,270 @@
+package cloudapinic
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	ionoscloud "github.com/ionos-cloud/sdk-go/v6"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services/cloudapi"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils"
+)
+
+type Service struct {
+	Client *ionoscloud.APIClient
+	Meta   interface{}
+	D      *schema.ResourceData
+}
+
+func (fs *Service) Get(ctx context.Context, datacenterID, serverID string, depth int32) ([]ionoscloud.Nic, error) {
+	nics, apiResponse, err := fs.Client.NetworkInterfacesApi.DatacentersServersNicsGet(ctx, datacenterID, serverID).Depth(depth).Execute()
+	apiResponse.LogInfo()
+	if err != nil {
+		return nil, err
+	}
+	if apiResponse.HttpNotFound() || nics.Items == nil || len(*nics.Items) == 0 {
+		log.Printf("[DEBUG] no nics found for datacenter %s, server %s", datacenterID, serverID)
+		return nil, nil
+	}
+	return *nics.Items, nil
+}
+
+func (fs *Service) FindById(ctx context.Context, datacenterId, serverId, ID string, depth int32) (*ionoscloud.Nic, *ionoscloud.APIResponse, error) {
+	nic, apiResponse, err := fs.Client.NetworkInterfacesApi.DatacentersServersNicsFindById(ctx, datacenterId, serverId, ID).Depth(depth).Execute()
+	apiResponse.LogInfo()
+	if err != nil {
+		if apiResponse.HttpNotFound() {
```

Comment:
@cristiGuranIonos for other services we have this format:
```
func (c *Client) GetNodePool(ctx context.Context, clusterId, nodePoolId string) (dataplatform.NodePoolResponseData, *dataplatform.APIResponse, error) {
	cluster, apiResponse, err := c.sdkClient.DataPlatformNodePoolApi.GetClusterNodepool(ctx, clusterId, nodePoolId).Execute()
	apiResponse.LogInfo()
	return cluster, apiResponse, err
}
```
At first sight, it makes more sense to handle errors and to print messages inside the service method, so let's leave as it is but maybe in the future we will have the same format for services. I wrote this comment just as a note in order to inform you about this.

Reply:
I think `List` for all elements and `Get`  for a find by X are good names. Will rename